src/compiler/checker.ts

```diff
@@ -4022,7 +4022,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isSyntacticDefault(node: Node) {
-        return ((isExportAssignment(node) && !node.isExportEquals) || hasSyntacticModifier(node, ModifierFlags.Default) || isExportSpecifier(node));
+        return ((isExportAssignment(node) && !node.isExportEquals)
+            || hasSyntacticModifier(node, ModifierFlags.Default)
+            || isExportSpecifier(node)
+            || isNamespaceExport(node));

```

Comment:
Is the following covered under these rules?

```ts
import * as yadda from "./yadda";
export { yadda as default };
```

Reply:
Yes (`isExportSpecifier`)

Reply:
This function is only called when the symbol attached to the declaration was reached by the name `default`, which is why these don’t check specifically check for defaultness.

Reply:
It’s intended to distinguish between these ES module syntaxes and something like

```ts
export = {
  options: {},
  default: {},
}
```